server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3085,7 +3118,7 @@ SnapshotsInProgress computeUpdatedState() {
                     changedCount,
                     startedCount
                 );
-                return updated;
+                return updated.withUpdatedNodeIdsForRemoval(initialState);
             }
             return existing;
```

Comment:
Seeing how `UpdateNodeIdsToRemoveTask` updates `nodeIdsToRemove`, i.e. read `SnapshotsInProgress` from clusterState then calls the `SnapshotsInProgress. withUpdatedNodeIdsForRemoval` with the same cluster state, I wonder whether it is worthwhile to have an assertion here to ensure `existing` and `existing.withUpdatedNodeIdsForRemoval(initialState)` are identical.

Reply:
I don't think that holds true in general, we might not have run `UpdateNodeIdsForRemovalTask` yet.